qannotate/src/au/edu/qimr/qannotate/modes/TandemRepeatMode.java

```diff
@@ -140,11 +154,13 @@ public void addAnnotation(String dbfile) throws IOException {
 		int totalRepeat = 0; 
 		int totalBlock = 0;					
 		Map<String, BlockIndex> indexedBlock  = new HashMap<>();
-		for(String chr: repeats.keySet()){
-			 logger.debug("indexing blocks for " + chr);
-			 indexedBlock.put(chr,  makeIndexedBlock( repeats.get(chr)));
-			 totalRepeat += repeats.get(chr).size();
-			 totalBlock += indexedBlock.get(chr).index.size();	 
+		
```

Comment:
For my own education — is it considered better form to iterate over the entries and get the keys, rather than the keys themselves?

Reply:
my understanding is that when you iterate over the entryset, you are effectively performing a linear traversal over the elements in the map. KeySet does the same, but you are then adding in lookups on the map (2 in this instance)

Reply:
Of course, thanks. I wasn't seeing the calls to `repeats.get` in there.